common/src/main/java/org/sonarlint/intellij/common/util/SonarLintUtils.java

```diff
@@ -78,16 +79,20 @@ public static <T> T getService(@NotNull Module module, Class<T> clazz) {
     return t;
   }
 
+  public static boolean isCodeScanCloudAlias(@Nullable String url) {
+    if (url.contains("codescan.io")) {
```

Comment:
Can be simplified to `return url.endsWith(".codescan.io")`

Reply:
Also, it's wise to use a property allowing to bypass this check. I see in previous implementation it was possible to do it through `sonarlint.internal.sonarcloud.url` system property (see line 53).

Because even on our environments we have different domains already which does not contain `codescan.io` in URL, for example: `https://amazon-codescan-ecs.codescan-dev.de/`

And Amazon company has their own domains where they deploy our codescan product.

Reply:
@borisbsu - Updated the logic to support custom codescan url. Please verify the changes.